odoo/tools/cache.py

```diff
@@ -101,8 +102,12 @@ def lookup(self, method, *args, **kwargs):
             return r
         except KeyError:
             counter.miss += 1
-            value = d[key] = self.method(*args, **kwargs)
-            return value
+            #with args[0].env.registry._caches_lock:
+            with args[0].pool._Registry__caches_lock:  # to check with ci/security
+                with contextlib.suppress(KeyError):
+                    return d[key]
+                value = d[key] = self.method(*args, **kwargs)
+                return value
```

Comment:
Suggestion for the consistency of `counter`:
```suggestion
            with args[0].pool._Registry__caches_lock:  # to check with ci/security
                with contextlib.suppress(KeyError):
                    r = d[key]
                    counter.hit += 1
                    return r
                counter.miss += 1
                value = d[key] = self.method(*args, **kwargs)
                return value
```